packages/esm-patient-registration-app/src/patient-registration/patient-registration-utils.ts

```diff
@@ -192,10 +192,48 @@ export function getPatientIdentifiersFromFhirPatient(patient: fhir.Patient): Arr
   });
 }
 
-export function getPhonePersonAttributeValueFromFhirPatient(patient: fhir.Patient) {
+export async function getIdentifierFieldValuesFromFhirPatient(
+  patient: fhir.Patient,
+  identifierConfig,
+): Promise<{ [identifierFieldName: string]: PatientIdentifierValue }> {
+  const identifiers: FormValues['identifiers'] = {};
+
+  for (const identifier of patient.identifier) {
+    for (const config of identifierConfig) {
+      const identifierConfig = config.identifierTypeSystem === identifier.system ? config : null;
+
+      if (identifierConfig) {
+        let identifierTypeName;
+
+        const url = `${restBaseUrl}/patientidentifiertype/${identifierConfig.identifierTypeUuid}`;
+        await openmrsFetch(url).then((response) => {
```

Comment:
`await` here is a bad idea. It makes looping through each identifier make one request to the backend and then wait until processing the next one. It would be better to do something like:

```ts
// before for loop
const promises = [];

// around here
promises.push(openmrsFetch(url).then(response => {
   if (!response.data?.name) {
      return;
   }
   
   identifiers[response.data.name] = {
      // blah, blah
   }
}));